src/main/java/com/ericsson/ei/controller/FailedNotificationControllerImpl.java

```diff
@@ -60,27 +64,38 @@
     public ResponseEntity<?> getFailedNotifications(
             @RequestParam(value = "subscriptionName", required = true) final String subscriptionName,
             final HttpServletRequest httpRequest) {
-        ObjectMapper mapper = new ObjectMapper();
-        QueryResponse queryResponse = new QueryResponse();
-        QueryResponseEntity queryResponseEntity = new QueryResponseEntity();
-        try {
-            List<String> response = processMissedNotification.processQueryMissedNotification(subscriptionName);
-            if (!response.isEmpty()) {
-                queryResponseEntity = mapper.readValue(response.get(0), QueryResponseEntity.class);
+        Set<String> subscriptionNameList = new HashSet<>(
+                Arrays.asList(subscriptionName.split(",")));
+        JSONArray foundArray = new JSONArray();
+        List<String> notFoundList = new ArrayList<>();
+
+        subscriptionNameList.forEach(name -> {
```

Comment:
Why not use a list of Strings instead of a Set and use for (String subName : Subscriptions) {} to improve readability?

Reply:
This code is taken from subscriptions controller and I don't know why the List is needlessly wrapped around a hash. The list is already made into a list... I'll remove this extra step but the code in subscription controller will still look like this.

Reply:
Try to improve code when you write and do not tuplicate bad code :hammer: :wrench: